### ISSUE TITLE
show current location

### ISSUE THREAD

**Issue by derhuerst** [rastapasta/mapscii#2]
Would like to be able to center the map around my current location. I wrote [`location`](https://github.com/derhuerst/location) for this.

Will submit a PR if you don't mind.

**Comment by rastapasta** [comment]
lovely idea, go ahead!

**Comment by derhuerst** [comment]
would you like to have a specific key to query the location? what about showing a progress indicator?

**Comment by derhuerst** [comment]
[on the `current-location` branch, it will now query the current location after startup](https://github.com/derhuerst/mapscii/commits/current-location). rendering it is still missing.

**Comment by rastapasta** [comment]
> would you like to have a specific key to query the location?

like the idea, choose any one that feels good for you :)

> what about showing a progress indicator?

👍 [_notify](https://github.com/rastapasta/mapscii/blob/master/src/Mapscii.coffee#L187) will help
would love to include a simple UI at some point to display information like progress/meta/search/*

> what about setting the zoom level?

hum..., maybe on initial load keeping the minimal zoom, when pressing the corresponding key street/block-level-ish?

thanks for your work!

**Comment by derhuerst** [comment]
>> would you like to have a specific key to query the location?

> like the idea, choose any one that feels good for you :)

i picked `l`

`current-location` will show a notification now.

**Comment by rastapasta** [comment]
:heart: tried and works beautifully!

just one question came to my mind:
in terms of user control & privacy, should we maybe ask before sending the related data?

we could / i would implement a small info-footer with all keys on startup - with [l] being the first.

thoughts?

**Comment by derhuerst** [comment]
> in terms of user control & privacy, should we maybe ask before sending the related data?

good point. macOS will automatically ask if the helper tool (that `location` uses underneath) should be able to access your location. on Linux and Windows, this warning is currently missing.

from the perspective of `location`, we have two choices:

- let consuming tools such as `mapscii` ask for permission. this is easy to implement, as the prompt can be shown in the best context possible (web, cli, gui)
- let `location` ask for permission. this would have the advantage that we can use native system prompts.
